R/comparePlot.R

```diff
@@ -142,72 +135,66 @@ comparePlot <- function(compareDF,
-    if (is.null(compareDF[["xp"]]) | is.null(compareDF[["yp"]])) {
-        CompPlot = ggplot(compareDF, aes_string(x = x, y = y)) +
-            geom_point(shape = 1,
-                       size = symbolSize[["xUnique"]],
-                       color = symbolFill[["xUnique"]],
-                       fill = symbolFill[["xUnique"]],
-                       alpha = alpha) +
-            coord_equal(xlim = c(-scalemax, scalemax), ylim = c(-scalemax, scalemax))
-        if (dens2D == TRUE) {
-            CompPlot = CompPlot + geom_density2d(color = symbolFill[["yUnique"]])
-        }
-    } else
-        # DELUXE PLOT: plot groups in different colors/shapes
-        if (!is.null(compareDF[["xp"]]) & !is.null(compareDF[["yp"]])) {
-            # Plot the subsets
-            xindx = compareDF[["xp"]] <= pThreshold
-            yindx = compareDF[["yp"]] <= pThreshold
-
-            # Boolean indexes to parse groups
-            bothindx = xindx & yindx
-            neitherindx = !xindx & !yindx
-            xindx = xindx & !bothindx # Unique to X
-            yindx = yindx & !bothindx # Unique to y
-
-            # Create group factor column in compareDF
-            compareDF$group = NA
-            compareDF$group[bothindx] = "Common"
-            compareDF$group[xindx] = "X Unique"
-            compareDF$group[yindx] = "Y Unique"
-            compareDF$group[neitherindx] = "Not Significant"
-            compareDF <- compareDF %>%
-                dplyr::left_join(ssc)
-            compareDF$group <- compareDF$group %>%
-                factor(levels = c("Common", "X Unique", "Y Unique", "Not Significant"))
-
-            # Set an order field to control order of plotting
-            # Plot order is 1) Not Significant plotted first, 2) randomly plot X and Y Unique
-            # then plot Common last
-            compareDF$order = sample.int(nrow(compareDF)) + # Add random order (seeded for reproducibility)
-                nrow(compareDF) * (compareDF$group == "Common") +  # Assign common a high value to sort last
-                -nrow(compareDF) * (compareDF$group == "Not Significant") # Assign NotSig group neg values to sort first
+    x <- make.names(colnames(compareDF)[1])
+    y <- make.names(colnames(compareDF)[2])
+    colnames(compareDF)[1:2] <- make.names(colnames(compareDF)[1:2])
+
+    if (!is.null(compareDF[["xp"]]) & !is.null(compareDF[["yp"]])) {
+        sigMeasurePlot <- TRUE
+        #create group factor column in compareDF
+        compareDF$group                                                                        <- "Not Significant"
+        compareDF$group[compareDF[["xp"]] <= pThreshold]                                       <- "X Unique"
+        compareDF$group[compareDF[["yp"]] <= pThreshold]                                       <- "Y Unique"
+        compareDF$group[(compareDF[["xp"]] <= pThreshold) & (compareDF[["yp"]] <= pThreshold)] <- "Common"
```

Comment:
use a regular logical And, never a bitwise one.  &&

Reply:
done

Reply:
Reverted this change as per https://3.basecamp.com/3775257/buckets/16965597/todos/3420654714

Reply:
So, we are still using the bitwise operator?

Reply:
I am going through comments and recommendation.